contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -1234,6 +1235,154 @@ uint64_t Filter::getMergedConfigId(ProcessorState& state) {
   return id;
 }
 
+void Filter::defineMetricCommon(uint32_t metric_type, absl::string_view name, uint32_t* metric_id) {
+  auto type = static_cast<MetricType>(metric_type);
+
+  Stats::StatNameManagedStorage storage(name, config_->metricStore().scope_->symbolTable());
+  Stats::StatName stat_name = storage.statName();
+  if (type == MetricType::Counter) {
+    auto id = config_->metricStore().nextCounterMetricId();
+    auto c = &config_->metricStore().scope_->counterFromStatName(stat_name);
+    config_->metricStore().counters_.emplace(id, c);
+    *metric_id = id;
+  } else if (type == MetricType::Gauge) {
+    auto id = config_->metricStore().nextGaugeMetricId();
+    auto g = &config_->metricStore().scope_->gaugeFromStatName(
+        stat_name, Stats::Gauge::ImportMode::Accumulate);
+    config_->metricStore().gauges_.emplace(id, g);
+    *metric_id = id;
+  } else { // (type == MetricType::Histogram) {
+    auto id = config_->metricStore().nextHistogramMetricId();
```

Comment:
nit: adding an `ASSERT(type == MetricType::Histogram)` here is better than the comment above.

Reply:
done, thanks.